File: Practice/React-Native/FullNativeModules/ios/MyModule.m
//
//  MyModule.m
//  FullNativeModules
//
//  Created by Vladislav Onatskyi on 9/26/22.
//

#import <Foundation/Foundation.h>

#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(MyModule, NSObject)

RCT_EXTERN_METHOD(asyncMultiply:
                 (int)a
                 b:(int)b
                  resolve: (RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject
                  )

RCT_EXTERN_METHOD(multiply:
                  (int)a
                  b:(int)b
                  callback:(RCTResponseSenderBlock)callback
                  )

RCT_EXTERN_METHOD(multiplyReturn:
                  (int)a
                  b:(int)b
                  )

+ (BOOL)multiplyReturn{
  return YES;
}

RCT_EXTERN_METHOD(sampleMethod:
                  (RCTPromiseResolveBlock)resolve
                  rejecter: (RCTPromiseRejectBlock)reject
                  )


@end

File: Practice/React-Native/FullNativeModules/ios/MyModule.swift
//
//  MyModule.swift
//  FullNativeModules
//
//  Created by Vladislav Onatskyi on 9/26/22.
//

import Foundation

@objc(MyModule)
class MyModule: NSObject {
  
  @objc
  func asyncMultiply (_ a: Int, b: Int, resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
    let result = a * b
    
    resolve(result)
  }
  
  @objc(multiply: b: callback: )
  func multiply(a:Int, b:Int, callback:RCTResponseSenderBlock) {
      let result = a * b
    
      print([result])
      
      callback([result])
    }
  
  @objc(multiplyReturn: b: )
  func multiplyReturn(a:Int, b:Int) -> Bool {
      let result = a * b
    
      print([result])
      
      return true
    }
  
  @objc(sampleMethod: reject: )
  func sampleMethod(_ resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
    resolve(13)
  }
  
  
}
